tests/tracing/test_otel_logging.py

```diff
@@ -0,0 +1,223 @@
+    )
+
+
+def test_otel_endpoint_requires_experiment_id_header(mlflow_server):
+    """
+    Test that the OTel endpoint requires experiment ID header.
+    """
+    # Create protobuf request
+    span = OTelProtoSpan()
+    span.trace_id = bytes.fromhex("0000000000000002" + "0" * 16)
+    span.span_id = bytes.fromhex("00000002" + "0" * 8)
+    span.name = "test-span-no-header"
+
+    scope = InstrumentationScope()
+    scope.name = "test-scope"
+
+    scope_spans = ScopeSpans()
+    scope_spans.scope.CopyFrom(scope)
+    scope_spans.spans.append(span)
+
+    resource = Resource()
+    resource_spans = ResourceSpans()
+    resource_spans.resource.CopyFrom(resource)
+    resource_spans.scope_spans.append(scope_spans)
+
+    request = ExportTraceServiceRequest()
+    request.resource_spans.append(resource_spans)
+
+    response = requests.post(
+        f"{mlflow_server}/v1/traces",
+        data=request.SerializeToString(),
+        headers={"Content-Type": "application/x-protobuf"},
+        timeout=10,
+    )
+
+    assert response.status_code == 422
+
+
+def test_invalid_otel_span_format_returns_400(mlflow_server):
+    """
+    Test that invalid OpenTelemetry protobuf format returns HTTP 400.
+    """
+    # Send completely invalid protobuf data
+    invalid_protobuf_data = b"this is not valid protobuf data at all"
+
+    response = requests.post(
+        f"{mlflow_server}/v1/traces",
+        data=invalid_protobuf_data,
+        headers={
+            "Content-Type": "application/x-protobuf",
+            MLFLOW_EXPERIMENT_ID_HEADER: "test-experiment",
+        },
+        timeout=10,
+    )
+
+    assert response.status_code == 400, f"Expected 400, got {response.status_code}"
+
+
+def test_missing_required_span_fields_returns_422(mlflow_server):
+    """
+    Test that spans that fail MLflow conversion return HTTP 422.
+    """
+    # Create protobuf request with missing trace_id (this should cause MLflow conversion to fail)
+    span = OTelProtoSpan()
+    # Don't set trace_id - this should cause _from_otel_proto to fail
+    span.span_id = bytes.fromhex("00000001" + "0" * 8)
+    span.name = "incomplete-span"
+
+    scope = InstrumentationScope()
+    scope.name = "test-scope"
+
+    scope_spans = ScopeSpans()
+    scope_spans.scope.CopyFrom(scope)
+    scope_spans.spans.append(span)
+
+    resource = Resource()
+    resource_spans = ResourceSpans()
+    resource_spans.resource.CopyFrom(resource)
+    resource_spans.scope_spans.append(scope_spans)
+
+    request = ExportTraceServiceRequest()
+    request.resource_spans.append(resource_spans)
+
+    response = requests.post(
+        f"{mlflow_server}/v1/traces",
+        data=request.SerializeToString(),
+        headers={
+            "Content-Type": "application/x-protobuf",
+            MLFLOW_EXPERIMENT_ID_HEADER: "test-experiment",
+        },
+        timeout=10,
+    )
+
+    assert response.status_code == 422, f"Expected 422, got {response.status_code}"
```

Comment:
```suggestion
    assert response.status_code == 422
```

we can remove this. pytest tells us what `response.status_code` is

Reply:
Totally, done!